### PR TITLE
But what if I don't want to kill people? Disabler SMG Port.

### PR DESCRIPTION
**Pull request by ItsMarmite**
## What Does This PR Do
This PR adds a new weapon (and associated supply crates/projectiles) to the game the WT-450 Disabler SMG. A rapid fire (weaker hitting) disabler that devastates in open hallways.

The Disabler SMG fires weakened disabler projectiles that deal 15 Stamina damage per hit and incur a short eye blur effect, compared to the 30 stamina per hit of normal disablers.

The Disabler SMG itself is also slightly inaccurate, and will often fire shots to the left or right of the targeted area. Friendly fire is likely, if used in a small area within a  team setting without care.

The Disabler SMG fits in a bag, but cannot be holstered and requires two hands to actively fire.

The Disabler SMG can be purchased for 550 credits from Supply consoles from WT-450 Disabler SMG crates, each crate containing two Disabler SMGs each.

## Why It's Good For The Game
There's very little in terms of non lethal escalation that Security has outside of tranquiliser darts- or asking botany/scichem for non lethal chemicals such as tirizene, ether, further haloperidol. I'm hoping to add this so Security has additional options to order from Cargo that don't involve beating people to death or breaking every bone in their body with a ballistic rifle. Antistun/associated armours will still nullify the effectiveness of the disabler SMG and it primarily thrives in open hallways (preferably not filled with crew). The gun itself will likely struggle to be used in maintenance effectively unless in point blank range (in which case, you're probably using a shotgun, baton or other more effective close range weaponry.)

Outside of this? I'm sure cargo techs will occasionally rob them, as is typical of them to do and who doesn't like Cargo to be sufficiently armed.

## Images of changes
image
image

https://github.com/ParadiseSS13/Paradise/assets/85680653/33f20d47-b4b5-4a80-b514-b2c8b6e318b3

Original PR from TG Station: https://github.com/tgstation/tgstation/pull/79158
Sprites and sounds from TG station. Credit to DrDiasyl.

## Testing
I shot so, so many skrell, ordered a crate, unlocked the crate. Spent a good while tweaking numbers, I'm sure I'll end up tweaking them some more depending on feedback.

## Changelog
:cl:
add: Disabler SMG crates can now be purchased from Cargo, containing Disabler SMGs a new type of two handed rapid fire disabler.
/:cl:

### CODE REVIEW

**Comment by Henri215** [comment]
Good idea to ask Octus for a extended description for this gun later, since he has a PR open for all energy weapons.

File: code/modules/projectiles/ammunition/energy_lens.dm [hunk 1]
```diff
@@ -165,6 +165,14 @@
 	fire_sound = 'sound/weapons/taser2.ogg'
 	harmful = FALSE
 	delay = 0.6 SECONDS
+/obj/item/ammo_casing/energy/disabler/smg
```
Comment (Henri215) [inline code/modules/projectiles/ammunition/energy_lens.dm:168]:
```suggestion

/obj/item/ammo_casing/energy/disabler/smg
```

File: code/modules/projectiles/guns/energy/stun.dm [hunk 2]
```diff
@@ -1,6 +1,7 @@
 /obj/item/gun/energy/taser
 	name = "taser gun"
 	desc = "A small, low capacity gun used for non-lethal takedowns."
+	weapon_weight = WEAPON_HEAVY
```
Comment (Henri215) [inline code/modules/projectiles/guns/energy/stun.dm:4]: Giving heavy weight to the taser gun?

File: code/modules/projectiles/projectile/beams.dm [hunk 3]
```diff
@@ -97,7 +97,13 @@
 	hitsound = 'sound/weapons/tap.ogg'
 	eyeblur = 0
 	impact_effect_type = /obj/effect/temp_visual/impact_effect/blue_laser
-	light_color = LIGHT_COLOR_CYAN
+	light_color = LIGHT_COLOR_LIGHT_CYAN
```
Comment (Henri215) [inline code/modules/projectiles/projectile/beams.dm:100]: Worth adding to the PR description that the disabler beam color is now slightly different.

Comment (DGamerL) [reply code/modules/projectiles/projectile/beams.dm:100]: Why are we even changing it for normal disablers?

File: code/modules/projectiles/guns/energy/stun.dm (hunk 2 above)
Comment (ItsMarmite) [reply code/modules/projectiles/guns/energy/stun.dm:4]: Not intentional, just a mistake I didn't notice. Correcting it in next commit.

File: code/modules/projectiles/projectile/beams.dm (hunk 3 above)
Comment (ItsMarmite) [reply code/modules/projectiles/projectile/beams.dm:100]: Yet another mistake from when I was testing something, thanks for catching it out. Changing it back in next commit.

File: code/__DEFINES/lighting_defines.dm [hunk 4]
```diff
@@ -33,7 +33,7 @@
 #define LIGHT_COLOR_WHITE		"#FFFFFF"
 #define LIGHT_COLOR_RED        "#FA8282" //Warm but extremely diluted red. rgb(250, 130, 130)
 #define LIGHT_COLOR_GREEN      "#64C864" //Bright but quickly dissipating neon green. rgb(100, 200, 100)
-#define LIGHT_COLOR_BLUE       "#6496FA" //Cold, diluted blue. rgb(100, 150, 250)
+#define LIGHT_COLOR_BLUE       "#6496fa" //Cold, diluted blue. rgb(100, 150, 250)
```
Comment (DGamerL) [inline code/__DEFINES/lighting_defines.dm:36]:
```suggestion
#define LIGHT_COLOR_BLUE       "#6496FA" //Cold, diluted blue. rgb(100, 150, 250)
```
Should just stay the same for consistency

File: code/modules/projectiles/guns/energy/stun.dm [hunk 5]
```diff
@@ -67,6 +67,25 @@
 
 	return ..()
 
+/obj/item/gun/energy/disabler/smg
+	name = "disabler smg"
+	desc = "An automatic disabler variant, as opposed to the conventional model, boasts a higher ammunition capacity at the cost of slightly reduced beam effectiveness."
+	icon_state = "disabler_smg"
+	item_state = null
```
Comment (DGamerL) [inline code/modules/projectiles/guns/energy/stun.dm:74]:
```suggestion
```
Already done on the parent type

File: code/modules/projectiles/guns/energy/stun.dm [hunk 6]
```diff
@@ -67,6 +67,25 @@
...
+	ammo_type = list(/obj/item/ammo_casing/energy/disabler/smg)
+	ammo_x_offset = 2
+	burst_size = 2
+	fire_delay = 2.5
+	shaded_charge = TRUE
+	can_flashlight = TRUE
+	can_holster = FALSE
+	flight_x_offset = 15
+	flight_y_offset = 10
+	execution_speed = 5 SECONDS
```
Comment (DGamerL) [inline code/modules/projectiles/guns/energy/stun.dm:87]:
```suggestion
	can_holster = FALSE
```
Also all defined on the parent type

File: code/modules/projectiles/guns/energy/stun.dm [hunk 7]
```diff
@@ -67,6 +67,25 @@
...
+/obj/item/gun/energy/disabler/smg
+	name = "disabler smg"
+	desc = "An automatic disabler variant, as opposed to the conventional model, boasts a higher ammunition capacity at the cost of slightly reduced beam effectiveness."
+	icon_state = "disabler_smg"
+	item_state = null
+	weapon_weight = WEAPON_HEAVY
+	w_class = WEIGHT_CLASS_BULKY
+	origin_tech = "combat=3"
+	ammo_type = list(/obj/item/ammo_casing/energy/disabler/smg)
+	ammo_x_offset = 2
```
Comment (DGamerL) [inline code/modules/projectiles/guns/energy/stun.dm:79]:
```suggestion
```

**Comment by Danchi299** [comment]
man, two disablers with price higher than 3 laser guns.
I wonder if it can be turned into RnD upgrade...

File: code/modules/projectiles/guns/energy/stun.dm [hunk 8]
```diff
@@ -67,6 +67,19 @@
 
 	return ..()
 
+/obj/item/gun/energy/disabler/smg
+	name = "disabler smg"
+	desc = "An automatic disabler variant, as opposed to the conventional model, boasts a higher ammunition capacity at the cost of slightly reduced beam effectiveness."
+	icon_state = "disabler_smg"
+	weapon_weight = WEAPON_HEAVY
+	w_class = WEIGHT_CLASS_BULKY
+	origin_tech = "combat=3"
```
Comment (Henri215) [inline code/modules/projectiles/guns/energy/stun.dm:76]:
```suggestion
```
It is inherited already.

**Review by Henri215 — COMMENTED** [review]
A flashlight can be attached to it? The sprite is positioned for it?

**Comment by ItsMarmite** [comment]
> A flashlight can be attached to it? The sprite is positioned for it?

PR already shows what it looks like with a flashlight attached, its the sprite of the gun on the floor.

**Comment by Henri215** [comment]
> > A flashlight can be attached to it? The sprite is positioned for it?
> 
> PR already shows what it looks like with a flashlight attached, its the sprite of the gun on the floor.

Took me a while to realize that was a flashlight, looking good then.

File: code/modules/projectiles/guns/energy/stun.dm [hunk 9]
```diff
@@ -67,6 +67,18 @@
 
 	return ..()
 
+/obj/item/gun/energy/disabler/smg
+	name = "disabler smg"
+	desc = "An automatic disabler variant, as opposed to the conventional model, boasts a higher ammunition capacity at the cost of slightly reduced beam effectiveness."
```
Comment (lewcc) [inline code/modules/projectiles/guns/energy/stun.dm:72]:
```suggestion
	desc = "An automatic disabler variant, as opposed to the conventional model. Boasts a higher ammunition capacity at the cost of slightly reduced beam effectiveness."
```

**Review by DGamerL — APPROVED** [review]
This thing is gonna be a lot of fun

**Review by S34NW — APPROVED** [review]

**Review by lewcc — APPROVED** [review]

**Pull request merged**